Detectors/TPC/calibration/include/TPCCalibration/CalibPadGainTracks.h

```diff
@@ -0,0 +1,521 @@
+    } else if (i == totalBins - 1 && mUseOverflow) {
+      std::cout.width(9);
+      std::cout << std::right << "O_Flow | "; // print Underflow bin in case of underflow bin is used
+    } else {
+      std::cout.width(6);
+      std::cout << std::fixed << std::setprecision(prec) << mXmin + binWidth * (i - mUseUnderflow) << " | ";
+    }
+
+    for (int j = 1; j <= getBinContent(i); ++j) {
+      std::cout << "x ";
+    }
+    std::cout << "\n";
+  }
+  std::cout << "\n";
+}
+
+template <class T>
+inline math_utils::math_base::StatisticsData FastHisto<T>::getStatisticsData(const float low, const float high) const
+{
+  math_utils::math_base::StatisticsData data{};
+  // in case something went wrong the COG is the histogram lower limit
+  data.mCOG = mXmin;
+  if (mBinCont.size() == 0) {
+    return data;
+  }
+  if (low > high) {
+    return data;
+  }
+
+  // total histogram content
+  const T integral = std::accumulate(mBinCont.begin(), mBinCont.end(), 0);
+  if (integral == 0) {
+    return data;
+  }
+  const float lowerBound = integral * low;
+  const float upperBound = integral * high;
+  const unsigned int lastBin = mBinCont.size() - 1;
+
+  float mean = 0;
+  float sum = 0;
+  float rms2 = 0;
+
+  // add fractional values
+  const float binWidth = getBinWidth();
+  const int shiftBin = mUseUnderflow ? -1 : 0; // shift the bin position in case of additional bin on the left
+
+  T countContent = 0;
+  bool isLowerFrac = false;
+  unsigned int bin = 0;
+  for (bin = 0; bin <= lastBin; ++bin) {
+    countContent += mBinCont[bin];
+
+    if (countContent - lowerBound <= 0) { // lower truncation cut
+      continue;
+    } else if (countContent >= upperBound) { // upper truncation cut
+      break;
+    }
+
+    const float xcenter = mXmin + (bin + 0.5f + shiftBin) * binWidth;
+    const float tmpBinCont = isLowerFrac ? mBinCont[bin] : countContent - lowerBound; // set bincontent to countcontent for first time only
+    const float tmpMean = tmpBinCont * xcenter;
+    mean += tmpMean;
+    sum += tmpBinCont;
+    rms2 += tmpMean * xcenter;
+    isLowerFrac = true;
+  }
+
+  if (!checkBin(bin)) {
+    return data;
+  }
+
+  //set last bin
+  //TODO move to upper loop
+  const float xcenter = mXmin + (bin + 0.5f + shiftBin) * binWidth;
+  const T upFrac = mBinCont[bin] - (static_cast<float>(countContent) - upperBound);
+  const float tmpMean = upFrac * xcenter;
+  mean += tmpMean;
+  sum += upFrac;
+  rms2 += tmpMean * xcenter;
+
+  if (sum == 0) {
+    return data;
+  }
+
+  mean /= sum;
+  data.mCOG = mean;
+  rms2 /= sum;
+  data.mStdDev = std::sqrt(std::abs(rms2 - mean * mean));
+  data.mSum = sum;
+
+  return data;
+}
+
+template <class T>
+inline int FastHisto<T>::findBin(const T val) const
+{
+  if (val < mXmin) {
+    if (!mUseUnderflow) {
+      std::cout << "WARNING IN FINDBIN()! UNDERFLOW BIN: BIN IS NOT IN HISTOGRAM RANGE!" << std::endl;
```

Comment:
I think it would be better to 
```c++
#include "Framework/Logger.h"
```
and use everywhere something like:
```c++
LOGP(warning, "formatted text", format args);
```
or in this case
```c++
LOGP(warning, "findBin: UNDERFLOW BIN: BIN IS NOT IN HISTOGRAM RANGE!");
```
also in other places, of course.

Reply:
Thanks, I changed all cout to LOGP()